Index: shortport.lua
===================================================================
--- shortport.lua	(revision 27084)
+++ shortport.lua	(working copy)
@@ -167,9 +167,13 @@
   Apache Tomcat HTTP server default ports: 8180 and 8000
   Litespeed webserver default ports: 8088 and 7080
 --]]
-http = shortport.port_or_service({80, 443, 631, 7080, 8080, 8088, 5800, 3872, 8180, 8000},
-	{"http", "https", "ipp", "http-alt", "vnc-http", "oem-agent"})
+local LIKELY_HTTP_PORTS = {80, 443, 631, 7080, 8080, 8088, 5800, 3872, 8180, 8000}
+local LIKELY_HTTP_SERVICES = {"http", "https", "ipp", "http-alt", "vnc-http", "oem-agent"}
 
+http = shortport.port_or_service(LIKELY_HTTP_PORTS,LIKELY_HTTP_SERVICES)
+http_service = shortport.service(LIKELY_HTTP_SERVICES)
+http_port = shortport.portnumber(LIKELY_HTTP_PORTS)
+
 local LIKELY_SSL_PORTS = {
     443, 465, 587, 636, 989, 990, 992, 993, 994, 995, 5061, 6679, 6697, 8443,
     9001,
